modules/watson-machine-learning/main.tf

```diff
@@ -10,6 +10,22 @@ module "cos" {
   cos_plan          = "standard"
 }
 
+module "storage_delegation" {
```

Comment:
The configure_project resource needs a dependency on this - the storage delegation should be set before creating a project. I think it would be better to include some output value from this module to the parameters of restapi_object.configure_project instead of explicit depends_on

Reply:
I've put it in as an explicit depends on, as if there's no storage delegation, the module won't be run and there will be no output. Let me know if there's a better way of handling this

Reply:
Usually I would just use some dummy expression with the module's output to establish an implicit dependency - it works better than the explicit one. In this case the `storage_delegation` module "conveniently" has no output, so you cannot even use that trick.
It probably should be ok in this case with just the explicit delegation.